File: BCNPA.cpp
#include<stdio.h>
#include<math.h>

int NT(long n)
{
	if(n<2)
		return 0;
	for(long h=2;h<=sqrt(n);h++)
		if(n%h==0)
			return 0;
	return 1;
}

int main(void)
{
	int T;
	scanf("%d",&T);
	while(T--)
	{
		long n,count=0;
		scanf("%ld",&n);
		if(NT(n))
			count++;
		int s=0;
		for(int i=2;i<=n/2;i++)
		{
			s=0;
			if(NT(i))
			{
				for(int j=i;j<=n;j++)
				{
					if(NT(j))
					{
						s+=j;
						if(s==n)
							count++;
						if(s>n)
							break;
					}
				}
			}
		}
		printf("%ld",count);
 		if(T!=0)
 			printf("\n");
	 }
 return 0;
}

File: spoj py/BCNPA.py
from sys import exit
from math import sqrt
def NT(n):
    if n<2:
        return 0
    for i in range(2,int(sqrt(n))+1,1):
        if n%i==0:
            return 0
    return 1

T = int(input())
while T>0:
    n = int(input())
    s = count = 0
    if NT(n)==1:
        count = count+1
    for i in range(2,n/2+1):
        s = 0
        if NT(i)==1:
            for j in range(i,n+1):
                if NT(j)==1:
                    s = s+j
                    if s==n:
                        count = count+1
                    if s>n:
                        break
    print count
    T=T-1
exit(0)
